crates/ra_db/src/input.rs

```diff
@@ -254,12 +254,12 @@ impl CrateGraph {
             return false;
         }
 
+        if target == from {
+            return true;
+        }
```

Comment:
Yeah, when writing dfses it always make sense to check conditions on entry, not before the recursive calls.... It bugs me a bit that a problem is known (as I've been bitten by this oh so many times), but that it slipped through code review... I guess, there's nothing actionable process-wise we can make better here :(

also cc @gfreezy for a "check condition on dfs *entry*" rule-of-thumb :) 

bors r+